adding check_install for quick_* files where necessary

reuning opened the pull request:
I added some lines to check if the requisite packages are available for the `quick_*()` functions. I also indicated in the documentation what packages were needed. 

(I was showing students huxtable in a bootcamp and the error thrown about needing to install packages threw them, this might be unnecessary but thought it would be helpful).

hughjonesd commented:
Could you alter this to use huxtable's `assert_package`? This will be more consistent. 

Arguably assert_package could be replaced everywhere by the rlang function. That would be a different PR though.

reuning commented:
Sure, I switched it over to `assert_package()` for now. I dropped the check for flextable as I noticed that `as_flextable()` already does that and is called before additional flextable functons are called.

hughjonesd reviewed (changes requested):
As the first argument to assert_package, use the name of calling function - the user won't know about e.g. `read_docx` but will know about `quick_docx`.

hughjonesd commented:
Don't worry about failing checks - that's not your fault

reuning commented:
That makes a lot more sense. Sorry about that. Given that I also added `assert_package()`calls for flextable to clarify why the user needs flextable. It might be unnecessary. Hopefully this is good to go now.

hughjonesd commented:
Thanks, merged!
